frontend/tests/badges/cppCourseGraduateTest.php

```diff
@@ -0,0 +1,138 @@
+            \OmegaUp\Controllers\Course::apiAddProblem(new \OmegaUp\Request([
+                'auth_token' => $adminLogin->auth_token,
+                'course_alias' => $courseData['course_alias'],
+                'assignment_alias' => $courseData['assignment_aliases'][1],
+                'problem_alias' => $problemData['request']['problem_alias'],
+            ]));
+        }
+
+        // Add students to course
+        $students = [];
+        for ($i = 0; $i < $studentsInCourse; $i++) {
+            $students[] = \OmegaUp\Test\Factories\Course::addStudentToCourse(
+                $courseData
+            );
+        }
+
+        $submissionSource = "#include <stdio.h>\nint main() { printf(\"3\"); return 0; }";
+        {
+            // This user will recive the badge because he will resolve 4 differents problems of the course
+            $studentLogin = \OmegaUp\Test\ControllerTestCase::login(
+                $students[0]
+            );
+
+            // Add one run to the first problem in the first assignment.
+            $runResponse = \OmegaUp\Controllers\Run::apiCreate(new \OmegaUp\Request([
+                'auth_token' => $studentLogin->auth_token,
+                'problemset_id' => $courseData['assignment_problemset_ids'][0],
+                'problem_alias' => $problems[0]['problem']->alias,
+                'language' => 'c11-gcc',
+                'source' => $submissionSource,
+            ]));
+            \OmegaUp\Test\Factories\Run::gradeRun(
+                /*$runData=*/ null,
+                1,
+                'AC',
+                null,
+                $runResponse['guid']
+            );
+            $runResponse = \OmegaUp\Controllers\Run::apiCreate(new \OmegaUp\Request([
+                'auth_token' => $studentLogin->auth_token,
+                'problemset_id' => $courseData['assignment_problemset_ids'][0],
+                'problem_alias' => $problems[1]['problem']->alias,
+                'language' => 'c11-gcc',
+                'source' => $submissionSource,
+            ]));
+            \OmegaUp\Test\Factories\Run::gradeRun(
+                /*$runData=*/ null,
+                1,
+                'AC',
+                null,
+                $runResponse['guid']
+            );
+
+            // Add one run to the third problem in the second assignment
+            $runResponse = \OmegaUp\Controllers\Run::apiCreate(new \OmegaUp\Request([
+                'auth_token' => $studentLogin->auth_token,
+                'problemset_id' => $courseData['assignment_problemset_ids'][1],
+                'problem_alias' => $problems[2]['problem']->alias,
+                'language' => 'c11-gcc',
+                'source' => $submissionSource,
+            ]));
+            \OmegaUp\Test\Factories\Run::gradeRun(
+                /*$runData=*/ null,
+                1,
+                'AC',
+                null,
+                $runResponse['guid']
+            );
+
```

Comment:
```suggestion
}

{
```

así es más claro que son dos usuarios distintos.

alternativamente, será posible separarlo en dos funciones distintas?

Reply:
va lo voy a separar en dos funciones una que si le de el badge y una que no

Reply:
va! el código en común para ambas funciones se puede de-duplicar moviéndolo a [`setUp()`](https://phpunit.readthedocs.io/en/9.0/fixtures.html).